linkerd/app/src/lib.rs

```diff
@@ -40,6 +40,11 @@ pub struct Config {
     pub inbound: inbound::Config,
     pub gateway: gateway::Config,
 
+    // In "ingress mode", we assume we are always routing HTTP requests and do
+    // not perform per-target-address discovery. Non-HTTP connections are
+    // forwarded without discovery/routing/mTLS.
+    pub ingress_mode: bool,
```

Comment:
Yeah, this was put here intentionally because we may opt to change inbound behavior in the future; and, more importantly, nothing in the outbound crate needs to be aware of this flag currently.